src/idl_gen_swift.cpp

```diff
@@ -0,0 +1,719 @@
+    code_ += "public {{ENUM_NAME}}: {{BASE_TYPE}}, Enum { ";
+    code_ += "\tpublic typealias T = {{BASE_TYPE}}";
+    code_ +=
+        "\tpublic static var byteSize: Int { return "
+        "MemoryLayout<{{BASE_TYPE}}>.size "
+        "}";
+    code_ += "\tpublic var value: {{BASE_TYPE}} { return self.rawValue }";
+
+    std::string enum_code = "\tcase ";
+    int keyCount = 0;
+    for (auto it = enum_def.Vals().begin(); it != enum_def.Vals().end(); ++it) {
+      const auto &ev = **it;
+      auto key = "KEY" + NumToString(keyCount);
+      auto value = "VALUE" + NumToString(keyCount);
+      auto name = Name(ev);
+      LowerCase(&name);
+      code_.SetValue(key, name);
+      code_.SetValue(value, enum_def.ToString(ev));
+      enum_code += "{{" + key + "}} = {{" + value + "}}, ";
+      keyCount++;
+    }
+    code_ += enum_code.substr(0, enum_code.size() - 2);
+    code_ += "}\n";
+  }
+
+  void GenLookup(const FieldDef &key_field) {
+    code_.SetValue("OFFSET", NumToString(key_field.value.offset));
+    auto offset_reader =
+        "Table.offset(Int32(fbb.capacity) - tableOffset, vOffset: {{OFFSET}}, "
+        "fbb: fbb)";
+    std::string spacing = "\t\t";
+    std::string double_spacing = spacing + "\t";
+
+    code_.SetValue("TYPE", GenType(key_field.value.type));
+    code_ +=
+        "\tpublic static func lookupByKey(vector: Int32, key: {{TYPE}}, fbb: "
+        "ByteBuffer) -> {{VALUENAME}}? {";
+
+    if (key_field.value.type.base_type == BASE_TYPE_STRING)
+      code_ += spacing + "let key = key.utf8.map { $0 }";
+    code_ += spacing +
+             "var span = fbb.read(def: Int32.self, position: Int(vector - 4))";
+    code_ += spacing + "var start: Int32 = 0";
+    code_ += spacing + "while span != 0 {";
+    code_ += double_spacing + "var middle = span / 2";
+    code_ +=
+        double_spacing +
+        "let tableOffset = Table.indirect(vector + 4 * (start + middle), fbb)";
+    if (key_field.value.type.base_type == BASE_TYPE_STRING) {
+      code_ += double_spacing + "let comp = Table.compare(" + offset_reader +
+               ", key, fbb: fbb)";
+    } else {
+      code_ += double_spacing +
+               "let comp = fbb.read(def: {{TYPE}}.self, position: Int(" +
+               offset_reader + "))";
+    }
+
+    code_ += double_spacing + "if comp > 0 {";
+    code_ += double_spacing + "\tspan = middle";
+    code_ += double_spacing + "} else if comp < 0 {";
+    code_ += double_spacing + "\tmiddle += 1";
+    code_ += double_spacing + "\tstart += middle";
+    code_ += double_spacing + "\tspan -= middle";
+    code_ += double_spacing + "} else {";
+    code_ += double_spacing + "\treturn {{VALUENAME}}(fbb, o: tableOffset)";
+    code_ += double_spacing + "}";
+    code_ += spacing + "}";
+    code_ += spacing + "return nil";
+    code_ += "\t}";
+  }
+
+  std::string GenOffset() { return "let o = {{ACCESS}}.offset({{OFFSET}}); "; }
+
+  std::string GenReaderMainBody(std::string optional = "") {
```

Comment:
V813 Decreased performance. The 'optional' argument should probably be rendered as a constant reference. idl_gen_swift.cpp 560